### DISCUSSION TITLE
poetry install vs poetry env use

### DISCUSSION THREAD

**Discussion by doolio** [python-poetry/poetry#6691] [Q&A] [answered]
Both these commands can create a virtual environment with the latter also activating it immediately. I notice `poetry env use` also creates a `envs.toml` file in `{cache-dir}/virtualenvs`. Curious why `poetry install` does not or why it is not necessary?

**Comment by neersighted** [comment, marked as answer]
These commands do radically different things:
* `poetry install` creates the Poetry-managed environment if it doesn't already exist, and then installs the project + its dependencies inside
* `poetry env use` set the Python used for the Poetry-managed environment, creating it if it doesn't already exist. `envs.toml` is created to record what the 'current interpreter' for the environment is (as the user has explicitly selected one, instead of letting Poetry select the interpreter implicitly)
